BASE/src/Microsoft.ApplicationInsights/Channel/IAsyncFlushable.cs

```diff
@@ -0,0 +1,22 @@
+﻿// <copyright file="IAsyncFlushable.cs" company="Microsoft">
+// Copyright © Microsoft. All Rights Reserved.
+// </copyright>
+
+namespace Microsoft.ApplicationInsights.Channel
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Represents asynchronous flush for  to application insights.
+    /// </summary>
+    public interface IAsyncFlushable : IDisposable
+    {
+        /// <summary>
+        /// Flushes the in-memory buffer asynchronously.
+        /// </summary>
+        /// <returns>The task to await.</returns>
```

Comment:
I think we still need to close on the meaning of the task result. It looks like it can be false if there were no items to flush. Is that the intent? Is the idea that callers should call it in a loop until it returns false?
What I'm getting at is what are clients supposed to do with the returned value? My guess is there's nothing reasonable they can or need to do.
Should this simply return `Task` instead of `Task<bool>`?

Reply:
+1 on having a clear meaning of the task result (and what could a caller do with it).

This is what I'm thinking about the **semantic** - a `SUCCESS` return from the flush API indicates that the ownership of telemetry data emitted before the flush invocation has transferred outside the process; a `FAILED` return (or should we just throw exception rather than return?) indicates a failure to transfer the ownership (so the process still owns all or part of the telemetry).

For example, if the SDK cannot send the telemetry data to the ingestion (e.g. due to network unavailable) AND cannot persist the data locally (e.g. due to disk full or no access), the data ownership is still tight to the process life, flush should return failure (or throw).

Reply:
Added detailed documentation for return type